### ISSUE TITLE
Unable to deserialize null value for type `Optional[UUID] = None`

### ISSUE THREAD

**Issue by kmsquire** [yukinarit/pyserde#263]
Hi, pyserde seems to mishandle this very specific corner case.

Minimum example:

```python
from serde import serde, to_dict, from_dict
from uuid import UUID
from typing import Optional
from dataclasses import dataclass

@serde
@dataclass
class NameId:
    name: str
    id: Optional[UUID] = None  # <====== UUID and `None` here is the problem 

x = NameId("Fred", None)
j = to_dict(x)
y = from_dict(NameId, j)
```

Everything before the last line works fine.  The last line gives the error below.

If I remove the default for `id` (i.e., remove `= None`), it works fine.
If I change the type of `id` to `Optional[int] = None`, it works fine.

So it's specific to `Optional[UUID] = None`.

From the error, it seems like it's trying to parse `None` as a `UUID`.

Here's the error:
```
Traceback (most recent call last):
  File "/Users/kevin.squire/Library/Caches/pypoetry/virtualenvs/pyserde-Y2i1skxI-py3.10/lib/python3.10/site-packages/serde/de.py", line 338, in from_obj
    res = serde_scope.funcs[func_name](c, maybe_generic=maybe_generic, data=o, reuse_instances=reuse_instances)
  File "<string>", line 13, in from_dict
  File "/Users/kevin.squire/.pyenv/versions/3.10.5_x86/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/uuid.py", line 171, in __init__
    raise TypeError('one of the hex, bytes, bytes_le, fields, '
TypeError: one of the hex, bytes, bytes_le, fields, or int arguments must be given

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/kevin.squire/tmp/pyserde/tmp/test.py", line 15, in <module>
    y = from_dict(NameId, j)
  File "/Users/kevin.squire/Library/Caches/pypoetry/virtualenvs/pyserde-Y2i1skxI-py3.10/lib/python3.10/site-packages/serde/de.py", line 414, in from_dict
    return from_obj(cls, o, named=True, reuse_instances=reuse_instances)
  File "/Users/kevin.squire/Library/Caches/pypoetry/virtualenvs/pyserde-Y2i1skxI-py3.10/lib/python3.10/site-packages/serde/de.py", line 391, in from_obj
    raise SerdeError(e)
serde.compat.SerdeError: one of the hex, bytes, bytes_le, fields, or int arguments must be given
```

**Comment by kmsquire** [comment]
This seems to be the generated code.  It doesn't handle having an explicit null as a value in the object to be decoded.

```python
def from_dict(cls=cls, maybe_generic=None, data=None, reuse_instances=True):
    if reuse_instances is Ellipsis:
        reuse_instances = True

    if data is None:
        return None

    __name = data["name"]

    __id = (
        (
            (data["id"] if isinstance(data["id"], UUID) else UUID(data["id"]))
            if reuse_instances
            else UUID(data["id"])
        )
        if "id" in data
        else serde_scope.defaults["id"]
    )

    try:
        rv = cls(
            __name,
            __id,
        )
    except Exception as e:
        raise UserError(e)

    return rv
```

**Comment by kmsquire** [comment]
And if I remove the default value, I get the following for `__id`:

```python
    __id = (
        (
            (data["id"] if isinstance(data["id"], UUID) else UUID(data["id"]))
            if reuse_instances
            else UUID(data["id"])
        )
        if data.get("id") is not None
        else None
    )
```

So I think that it needs to be something like

```python
    __id = (
        (
            (data["id"] if isinstance(data["id"], UUID) else UUID(data["id"]))
            if reuse_instances
            else UUID(data["id"])
        )
        if data.get("id") is not None
        else None if "id" in data
        else serde_scope.defaults["id"]
    )
```

**Comment by kmsquire** [comment]
Fix available in #264

**Comment by yukinarit** [comment]
pyserde v0.9.2 was published in PyPI. Thank you!
